### PR TITLE
Adds RC-related parts to appMain and HMI message handler

### PR DESCRIPTION
**Pull request by dev-gh**
Partial RC logic implementation

### CODE REVIEW

**Comment by dev-gh** [comment]
@LuxoftAKutsan @AKalinich-Luxoft @LitvinenkoIra 
Please review

**Comment by dev-gh** [comment]
@AByzhynar
Please review

File: src/components/hmi_message_handler/include/hmi_message_handler/hmi_message_adapter_impl.h [hunk 1]
```diff
@@ -53,6 +53,14 @@ class HMIMessageAdapterImpl : public HMIMessageAdapter {
    */
   virtual ~HMIMessageAdapterImpl();
 
+#ifdef SDL_REMOTE_CONTROL
+  /**
+   * @brief Subscribes to notification from HMI
+   * @param hmi_notification string with notification name
+   */
+  virtual void SubscribeToHMINotification(
```
Comment (AKalinich-Luxoft) [inline src/components/hmi_message_handler/include/hmi_message_handler/hmi_message_adapter_impl.h]: @dev-gh I think `virtual` is redundant here if you already have `OVERRIDE`

File: src/components/hmi_message_handler/include/hmi_message_handler/hmi_message_handler_impl.h [hunk 2]
```diff
@@ -96,6 +96,14 @@ class HMIMessageHandlerImpl : public HMIMessageHandler,
   void AddHMIMessageAdapter(HMIMessageAdapter* adapter);
   void RemoveHMIMessageAdapter(HMIMessageAdapter* adapter);
 
+#ifdef SDL_REMOTE_CONTROL
+  /**
+   * @brief Subscribes to notification from HMI
+   * @param hmi_notification string with notification name
+   */
+  virtual void SubscribeToHMINotification(
```
Comment (AKalinich-Luxoft) [inline src/components/hmi_message_handler/include/hmi_message_handler/hmi_message_handler_impl.h]: @dev-gh same as above

File: src/components/hmi_message_handler/src/messagebroker_adapter.cc [hunk 3]
```diff
@@ -135,9 +136,18 @@ void MessageBrokerAdapter::SubscribeTo() {
   MessageBrokerController::subscribeTo("SDL.OnDeviceStateChanged");
   MessageBrokerController::subscribeTo("SDL.OnPolicyUpdate");
   MessageBrokerController::subscribeTo("BasicCommunication.OnEventChanged");
+  MessageBrokerController::subscribeTo("RC.OnDeviceLocationChanged");
```
Comment (AKalinich-Luxoft) [inline src/components/hmi_message_handler/src/messagebroker_adapter.cc]: @dev-gh according to proposal we should not subscribe to this notification

**Review by AKalinich-Luxoft — CHANGES_REQUESTED** [review]

File: src/components/hmi_message_handler/include/hmi_message_handler/hmi_message_adapter_impl.h (hunk 1 above)
Comment (dev-gh) [reply src/components/hmi_message_handler/include/hmi_message_handler/hmi_message_adapter_impl.h]: @AKalinich-Luxoft 
Done with 24d4f7a92d

File: src/components/hmi_message_handler/include/hmi_message_handler/hmi_message_handler_impl.h (hunk 2 above)
Comment (dev-gh) [reply src/components/hmi_message_handler/include/hmi_message_handler/hmi_message_handler_impl.h]: @AKalinich-Luxoft 
Done with 24d4f7a92d

File: src/components/hmi_message_handler/src/messagebroker_adapter.cc (hunk 3 above)
Comment (dev-gh) [reply src/components/hmi_message_handler/src/messagebroker_adapter.cc]: @AKalinich-Luxoft 
Done with 24d4f7a92d

**Review by AKalinich-Luxoft — APPROVED** [review]

**Review by LitvinenkoIra — APPROVED** [review]

**Review by LuxoftAKutsan — APPROVED** [review]

**Review by AByzhynar — APPROVED** [review]

**Pull request merged**